controllers/agentmachine_controller.go

```diff
@@ -408,10 +434,7 @@ func (r *AgentMachineReconciler) updateAgentMachineWithFoundAgent(ctx context.Co
 	return nil
 }
 
-func (r *AgentMachineReconciler) updateFoundAgent(ctx context.Context, log logrus.FieldLogger,
```

Comment:
Split the second half of this function into `bindAgent` since we've seen issues where the agent gets the clusterdeployment ref but the agentMachine doesn't actually exist or is getting deleted.

Reply:
No longer relevant